### PR TITLE
[RFC] Todo due date spec

### PR DESCRIPTION
**Pull request by scalvert**
[Rendered](https://github.com/ember-template-lint/ember-template-lint-todo-utils/blob/renatoi/master/docs/todo-due-dates.md)

### CODE REVIEW

File: docs/todo-due-dates.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+```json
+{
+  "lintTodo": {
+    "daysToDecay": {
+      "warn": 30,
+      "error": 90
+    }
+  }
+}
+```
```
Comment (MelSumner) [inline docs/todo-due-dates.md:71]: Are these warn/error numbers correct?

Comment (scalvert) [reply docs/todo-due-dates.md:71]: In what sense? They're examples, so I'm not sure what correctness you're referring to.

Comment (rwjblue) [reply docs/todo-due-dates.md:71]: These are just examples, there will not be any default value (from the OSS side) but if folks want to make their own linting setup that sets defaults this is how they'd do it.

File: docs/todo-due-dates.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+**Via config:**
+
+```json
+{
+  "lintTodo": {
+    "daysToWarn": 20,
+    "daysToError": 25
+  }
+}
```
Comment (rwjblue) [inline docs/todo-due-dates.md:99]: needs to use the new format 

```
  "lintTodo": {
    "daysToDecay": {
      "warn": 30,
      "error": 90
    }
  }
```

File: docs/todo-due-dates.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+[requirements]: #implementation
+
+## High Level Flow
+
+When an author runs a linter with the todo functionality, all errors are converted to todos with no due date threshold unless one is present in `package.json` _or_ passed as a command line argument.
+
+- `daysToWarn` - number of days after its creation date that a todo transitions into a `warn`
+- `daysToError` number of days after its creation date that a todo transitions into an `error`.
+- `daysToWarn` _and_ `daysToError` - if used, `daysToError` must be greater than `daysToWarn`.
```
Comment (rwjblue) [inline docs/todo-due-dates.md:46]: Lets redo this a tad bit to make it clear what the general concepts are, and then how you configure them in the two locations (package.json and command line)

File: docs/todo-due-dates.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+```
+
+**Via `ember-template-lint` CLI:**
+
+```shell
+$ yarn ember-template-lint . --update-todo --todo-days-to-warn 20
+
+Error: `--todo-days-to-warn` must be used with `--todo-days-to-error`.
+```
+
```
Comment (rwjblue) [inline docs/todo-due-dates.md:153]: We removed this constraint, this section should be removed

**Review by rwjblue — APPROVED** [review]

**Pull request merged**